internal/state/state_skel.go

```diff
@@ -454,3 +475,125 @@ func (s *stateSkel) checkAttributesExist(attrs nodeinfo.NodeAttributes, attrType
 	}
 	return nil
 }
+
+// isDaemonSetReadyOnDelete checks if a DaemonSet with OnDelete update strategy is ready
+// by verifying pod revision hashes and node placement
+func (s *stateSkel) isDaemonSetReadyOnDelete(ctx context.Context, ds *appsv1.DaemonSet, reqLogger logr.Logger) (bool, error) {
+	ownedPods, err := s.getOwnedPods(ctx, ds)
+	if err != nil {
+		reqLogger.Error(err, "Failed to list pods for DaemonSet")
+		return false, err
+	}
+	if len(ownedPods) == 0 {
+		return false, nil
+	}
+
+	if !s.arePodsHealthy(ownedPods) {
+		return false, nil
+	}
+
+	dsRevisionHash, err := s.getLatestRevisionHash(ctx, ds)
+	if err != nil {
+		// If revision hash unavailable, fall back to health-only check
+		reqLogger.V(consts.LogLevelWarning).Info("Could not get revision hash, using health-only check", "error", err)
+		return true, nil
+	}
+
+	// Check if all pods have the latest revision
+	for _, pod := range ownedPods {
+		if hash, ok := pod.Labels["controller-revision-hash"]; !ok || hash != dsRevisionHash {
+			// Pods have outdated revision - verify they're on nodes matching current nodeSelector
+			reqLogger.V(consts.LogLevelInfo).Info("Pods have outdated revision, verifying node placement")
+			return s.verifyNodePlacement(ctx, ds, ownedPods, reqLogger)
```

Comment:
Is node placement is the actual thing we care about for "ready" status? If so, can we just not check for node placement regardless of revision hash on the pods?

I don't think we necessarily need to know if a pod was updated in a level-triggered reconciliation. We just need to periodically check if the final condition is true.